File: lib/readaloud/readaloud_constants.dart
class ReadAloudConstants {
  ReadAloudConstants._();

  static const String channelName = 'readaloud';

  static const String methodSpeak = 'speak';
  static const String methodStop = 'stop';
  static const String methodIsSpeaking = 'isSpeaking';
  static const String methodDispose = 'dispose';

  static const String paramText = 'text';

  static const String errorInvalidArguments = 'INVALID_ARGUMENTS';
  static const String errorTtsNotInitialized = 'TTS_NOT_INITIALIZED';
  static const String errorSpeakFailed = 'SPEAK_FAILED';
  static const String errorException = 'EXCEPTION';

  static const String tooltipStop = 'Stop Reading';
  static const String tooltipRead = 'Read Aloud';
  static const String snackbarFailedStart = 'Failed to start reading';
  static const String snackbarNoDescription = 'No description available to read';
}

File: android/app/src/main/kotlin/com/example/news_reader/ReadAloudConstants.kt
package com.example.news_reader

object ReadAloudConstants {
    const val CHANNEL_NAME = "readaloud"

    const val METHOD_SPEAK = "speak"
    const val METHOD_STOP = "stop"
    const val METHOD_IS_SPEAKING = "isSpeaking"
    const val METHOD_DISPOSE = "dispose"

    const val PARAM_TEXT = "text"

    const val ERROR_INVALID_ARGUMENTS = "INVALID_ARGUMENTS"
    const val ERROR_TTS_NOT_INITIALIZED = "TTS_NOT_INITIALIZED"
    const val ERROR_SPEAK_FAILED = "SPEAK_FAILED"
    const val ERROR_EXCEPTION = "EXCEPTION"

    const val ERROR_MSG_INVALID_ARGUMENTS = "Text is required"
    const val ERROR_MSG_TTS_NOT_INITIALIZED = "TextToSpeech is not initialized"
    const val ERROR_MSG_SPEAK_FAILED = "Failed to speak text"

    const val UTTERANCE_ID = "utteranceId"
}